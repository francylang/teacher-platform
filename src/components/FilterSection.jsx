import React from 'react';
import PropTypes from 'prop-types';
import { PROD_URL } from '../utils/url.js';

const FilterSection = ({ renderFilteredDiscussions, renderDiscussions, domain }) => {
  const fetchDiscussions = standard => {
    fetch(`${PROD_URL}/api/v1/discussions`)
      .then((response) => response.json())
      .then((rawDiscussions) => findMatchingDiscussions(rawDiscussions, standard))
      .then((matchingDiscussions) => {
        renderDiscussions();
        renderFilteredDiscussions(matchingDiscussions);
      })
      .catch((error) => console.error({ error }));
  };

  const findMatchingDiscussions = (rawDiscussions, standard) => {
    return rawDiscussions.filter((discussion) => {
      if (standard === discussion.tagTitle) {
        return discussion;
      };
    });
  };

  const buildDomainList = () => {
    return domain.map((standard, index) => {
      let firstChar = standard.charAt(0);

      if (firstChar === '6' || firstChar === '7' || firstChar === '8') {
        return (
          <button
            key={index}
            className="standard-link"
            onClick={() => fetchDiscussions(standard)}
            key={standard}
          >
            <p className={`standard-text ${standard[2]}-link`}>
              {standard}
            </p>
          </button>
        );
      } else {
        return (
          <section
            className="domain-label-container"
            key={standard}
          >
            <h5 className="domain-label">
              {standard}
            </h5>
          </section>
        );
      }
    });
  };

  return (
    <aside className="domain-section">
      {buildDomainList()}
    </aside>
  );
};

FilterSection.propTypes = {
  domain: PropTypes.array,
  renderDiscussions: PropTypes.func,
  renderFilteredDiscussions: PropTypes.func,
};
export default FilterSection;
