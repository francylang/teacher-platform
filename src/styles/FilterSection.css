.domain-section {
  outline: 2px solid black;
  display: inline-block;
  width: 19%; }
