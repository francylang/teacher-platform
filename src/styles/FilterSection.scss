.domain-section {
  outline: 1px solid pink;
  display: inline-block;
  width: 19%;
}
