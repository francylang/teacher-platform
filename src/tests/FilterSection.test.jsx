import React from 'react';
import ReactDOM from 'react-dom';
import FilterSection from '../components/FilterSection';
import { shallow } from 'enzyme';
import testSetup from './testSetup';

describe('<FilterSection />', () => {
  let wrapper;
  let mockFn = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<FilterSection />);
  });

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should render a domain section', () => {
    const optionsContainer = wrapper.find('.domain-section');

    expect(optionsContainer.length).toEqual(1);
  });

  it('should render a button for each standard', () => {
    const btn = wrapper.find('.standard-link');

    expect(btn.length).toEqual(1);
  });

  it('should render text inside each button', () => {
    const text = wrapper.find('.standard-text');

    expect(text.length).toEqual(1);
  });

  it('should render a domain label container', () => {
    const container = wrapper.find('.domain-label-container');

    expect(container.length).toEqual(1);
  });

  it('should render a domain label', () => {
    const domainLabel = wrapper.find('.domain-label');

    expect(domainLabel.length).toEqual(1);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
})
