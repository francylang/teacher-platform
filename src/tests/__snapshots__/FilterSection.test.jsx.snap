// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterSection /> should match snapshot 1`] = `undefined`;
